.github/workflows/test-integration-rosa.yaml

```diff
@@ -0,0 +1,359 @@
+        uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3
+        with:
+          timeout_minutes: 10
+          max_attempts: 40
+          shell: bash
+          retry_wait_seconds: 15
+          command: |
+            : # see https://github.com/nick-fields/retry/issues/133
+            set -o errexit
+            set -o pipefail
+
+            cd ./.github/workflows/fixtures/clusters/rosa-hcp-on-demand/
+
+            : # configure external dns
+            export EXTERNAL_DNS_GCP_SERVICE_ACCOUNT=${{ secrets.DISTRO_CI_ON_DEMAND_EXTERNAL_DNS_GCP_SERVICE_ACCOUNT }}
+            envsubst < external-dns/secret.yaml.tpl > external-dns/secret.yaml
+            yq -ei '.txtOwnerId = "${{ fromJson(needs.read_cluster_vars.outputs.result).cluster_name[matrix.distro.name] }}"' external-dns/helm-chart/external-dns-values.yaml
+
+            : # configure ExternalSecretStore replication from the permanent ROSA cluster
+            export EXTERNAL_SECRET_STORE_SA_TOKEN=${{ secrets.DISTRO_CI_OPENSHIFT_EXTERNAL_SECRET_STORE_SA_TOKEN }}
+            export EXTERNAL_SECRET_STORE_SA_SERVICE_CA=${{ secrets.DISTRO_CI_OPENSHIFT_EXTERNAL_SECRET_STORE_SA_SERVICE_CA }}
+            export EXTERNAL_SECRET_STORE_SA_CA=${{ secrets.DISTRO_CI_OPENSHIFT_EXTERNAL_SECRET_STORE_SA_CA }}
+            envsubst < distribution-team/secret.yaml.tpl > distribution-team/secret.yaml
+            yq -ei '.spec.provider.kubernetes.server.url = "${{ secrets.DISTRO_CI_OPENSHIFT_CLUSTER_URL }}"' distribution-team/external-cluster-secretstore.yaml
+
+            : # apply!
+            kustomize build --enable-helm ./ | kubectl apply -f - 
+                
+            echo "Waiting for global state to converge"
+            sleep 30
+                
+            while kubectl get pods --all-namespaces | grep -E -q -v '(Running|Completed|STATUS)'; do
+                echo "Waiting for all pods to be Running or Completed"
+                sleep 5
+            done
+
+            kubectl get all --all-namespaces
+
+      - name: Export kubeconfig and encrypt it # this is required to pass matrix outputs securely using artifacts
+        id: export_kube_config
+        run: |
+          echo "$(kubectl config view --raw)" > kubeconfig.yaml 2>/dev/null
+          openssl enc -aes-256-cbc -salt -in kubeconfig.yaml -out encrypted_kubeconfig.enc -pass pass:"${GITHUB_TOKEN}" -pbkdf2
+          encrypted_kubeconfig_base64=$(base64 -w 0 encrypted_kubeconfig.enc)
+          echo "kubeconfig_raw=${encrypted_kubeconfig_base64}" >> "$GITHUB_OUTPUT"
+
+      ## Write for matrix outputs workaround
+      - uses: cloudposse/github-action-matrix-outputs-write@ed06cf3a6bf23b8dce36d1cf0d63123885bb8375 # v1
+        id: out
+        with:
+          matrix-step-name: ${{ github.job }}
+          matrix-key: ${{ matrix.distro.name }}
+          outputs: |-
+            kubeconfig_raw: ${{ steps.export_kube_config.outputs.kubeconfig_raw }}
```

Comment:
there's no input possible to define the days on how long the artifact should exist, or?
In the background they're using the Artifacts API to upload it to the GHA run.

Reply:
they're automatically deleted after I think 30 days or so.
It's fine, just a personal thing to delete such things after a day since they're just useful for that 1 execution.